osu.Game/Skinning/PoolableSkinnableSample.cs

```diff
@@ -102,7 +102,7 @@ private void updateSample()
             sampleContainer.Add(Sample = new DrawableSample(ch) { Looping = Looping });
 
             // Start playback internally for the new sample if the previous one was playing beforehand.
-            if (wasPlaying)
+            if (wasPlaying && !Looping)
```

Comment:
Upon actually checking, I actually don't think I originally misread and this should probably be

```suggestion
            if (wasPlaying && Looping)
```

right? The looping samples need to be restarted to continue playing, but the non-looping ones don't. It seems this logic can actually result in losing looping samples if I switch from skin A to B and then back to A again (i.e. during a spinner).

Reply:
It does indeed look back to front.

Reply:
In fact this actually might need to even be `wasPlaying || Looping` as I'm hitting some skins here that have `wasPlaying` set to apparently false despite being looping samples. I'm not yet 100% on what the cause of that would be, though.

Reply:
Never mind, the case I'm hitting is caused by some frivolous audio handling framework-side without error checking.